ocpmodels/datasets/datasets/gasdb.py

```diff
@@ -0,0 +1,268 @@
+                         for neighbors in structure.get_all_neighbors(r=self.radius, include_index=True)]
+
+        # Grab the distances and indices of each atoms' neighbors
+        all_distances, all_indices = [], []
+        dummy_distance, dummy_index = self.radius + 1, 0
+        for neighbors in all_neighbors:
+            try:
+                _, distances, indices, _ = list(zip(*neighbors))
+                distances = list(distances)
+                indices = list(indices)
+            # If there are no neighbors
+            except ValueError:
+                distances, indices = [dummy_distance], [dummy_index]
+
+            # Pad empty elements in the features
+            if len(distances) < self.max_num_nbr:
+                padding_length = self.max_num_nbr - len(distances)
+                distances.extend([dummy_distance] * padding_length)
+                indices.extend([dummy_index] * padding_length)
+
+            # Trim extra elements
+            elif len(distances) > self.max_num_nbr:
+                distances = distances[:self.max_num_nbr]
+                indices = indices[:self.max_num_nbr]
+
+            # Concatenation/formatting
+            all_distances.append(distances)
+            all_indices.append(indices)
+        all_distances, all_indices = np.array(all_distances), np.array(all_indices)
+
+        # Calculate the outputs
+        gaussian_distances = self.gdf.expand(all_distances)
+        embeddings = np.vstack([EMBEDDINGS[site.specie.number] for site in structure])
+
+        # Turn everything into tensors
+        embeddings = torch.Tensor(embeddings)
+        gaussian_distances = torch.Tensor(gaussian_distances)
+        all_indices = torch.LongTensor(all_indices)
+        return embeddings, gaussian_distances, all_indices
+
+
+class GaussianDistance:
+    """
+    Expands the distance by Gaussian basis.
+
+    Unit: angstrom
+    """
+    def __init__(self, dmin, dmax, step, var=None):
+        """
+        Parameters
+        ----------
+
+        dmin: float
+          Minimum interatomic distance
+        dmax: float
+          Maximum interatomic distance
+        step: float
+          Step size for the Gaussian filter
+        """
+        assert dmin < dmax
+        assert dmax - dmin > step
+        self.filter = np.arange(dmin, dmax + step, step)
+        if var is None:
+            var = step
+        self.var = var
+
+    def expand(self, distances):
+        """
+        Apply Gaussian disntance filter to a numpy distance array
+
+        Parameters
+        ----------
+
+        distance: np.array shape n-d array
+          A distance matrix of any shape
+
+        Returns
+        -------
+        expanded_distance: shape (n+1)-d array
+          Expanded distance matrix with the last dimension of length
+          len(self.filter)
+        """
+        return np.exp(-(distances[..., np.newaxis] - self.filter)**2 / self.var**2)
```

Comment:
This is same as https://github.com/Open-Catalyst-Project/baselines/blob/master/ocpmodels/datasets/gasdb.py?